easyrest/views/user_controller.py

```diff
@@ -110,3 +111,75 @@ def get_users_list(request):
                   request.dbsession.query(User).filter_by(role_id=derivable_role_id).order_by(User.name).all()]
 
     return wrap(users_list, success=True, message='Users with role \'{}\''.format(role.name))
+
+
+@view_config(route_name='user_create', renderer='json', request_method='POST')
+def create_user(request):
+    """This function is intended to create a user with a specific role.
+
+       This function processes the route /user/{role_id:\d+}
+       and tries to create a user depending on the identifier
+       that is extracted from the parameter {role_id}.
+
+    :param request: POST request
+    :return: If incorrect json:
+                {
+                  "message": null,
+                  "data": [],
+                  "success": false,
+                  "error": "Incorrect json format"
+                }
+             If validation errors:
+                {
+                  "message": null,
+                  "data": [],
+                  "success": false,
+                  "error": "validation_errors"
+                }
+             If user created successfully:
+                {
+                  "message": "User successfully added",
+                  "data": [],
+                  "success": true,
+                  "error": null
+                }
+    :raise HTTPNotFound: If user role id not found.
+    :raise HTTPForbidden: If if an authorization is needed to perform an action.
+    """
+    log = logging.getLogger(__name__)
+
+    try:
+        form_data = request.json_body
+    except ValueError as ve:
+        log.error(ve.message)
+        return wrap([], success=False, error='Incorrect json format')
+
+    database = request.dbsession
+    new_user_role = int(request.matchdict['role_id'])
+    role = database.query(UserRole).get(new_user_role)
+    if role is None:
+        log.error('Role id {} not found'.format(new_user_role))
+        raise HTTPNotFound(request.path)
+
+    if new_user_role == User.CLIENT:
+        try:
+            User.add(database, form_data)
+        except ValidationError as ve:
+            return wrap([], success=False, error=str(ve))
+        else:
+            return wrap([], success=True, message='User successfully added')
+
+    try:
+        current_user = request.token.user
+    except AttributeError as ae:
+        log.error(ae.message)
+        raise HTTPForbidden('Need token for access')
+
+    new_user_role_name = role.name
+    check_action_access(current_user.role.name, foreign_role=new_user_role_name, action='create')
+    try:
+        User.add(database, form_data, role=new_user_role)
+    except ValidationError as ve:
+        return wrap([], success=False, error=str(ve))
+    else:
+        return wrap([], success=True, message='User successfully added')
```

Comment:
Maybe can add this 2 try in some separate function because they are equal?

Reply:
Do you mean something like that:
   ```
 if new_user_role == User.CLIENT:
        return try_to_add_user(database, form_data)
       
    try:
        current_user = request.token.user
    except AttributeError as ae:
        log.error(ae.message)
        raise HTTPForbidden('Need token for access')

    new_user_role_name = role.name
    check_action_access(current_user.role.name, foreign_role=new_user_role_name, action='create')
    return try_to_add_user(database, form_data, new_user_role)


def try_to_add_user(database, form_data, role=User.CLIENT):
    try:
        User.add(database, form_data, role=role)
    except ValidationError as ve:
        return wrap([], success=False, error=str(ve))
    else:
        return wrap([], success=True, message='User successfully added')
```